lib/improver/tests/blending/weights/test_ChooseWeightsLinear.py

```diff
@@ -557,6 +558,58 @@ def test_spatial_varying_weights(self):
                                weights_cube.metadata)
 
 
+class Test__slice_input_cubes(IrisTest):
+    """Test the _slice_input_cubes method"""
+
+    def setUp(self):
+        """Set up plugin with suitable parameters (used for dict only)"""
+        self.plugin = ChooseWeightsLinear(
+            "forecast_period", config_dict=CONFIG_DICT_UKV)
+
+        # create a cube with unnecessary realization coordinate (dimensions:
+        # model_id: 2; realization: 1; time: 3; latitude: 2; longitude: 2)
+        cube = add_model_id_and_model_configuration(
+            set_up_temperature_cube(timesteps=3), promote_to_new_axis=True)
+        self.cube = add_forecast_reference_time_and_forecast_period(
+            cube, time_point=[412235.0, 412247.0, 412278.0],
+            fp_point=[8., 20., 51.])
+
+        # create a reference cube as above WITHOUT realization
+        new_data = self.cube.data[:, 0, :, :, :]
+        dim_coords = [(self.cube.coord("model_id"), 0),
+                      (self.cube.coord("time"), 1),
+                      (self.cube.coord("latitude"), 2),
+                      (self.cube.coord("longitude"), 3)]
+        aux_coords = [(self.cube.coord("model_configuration"), 0),
+                      (self.cube.coord("forecast_period"), 1),
+                      (self.cube.coord("forecast_reference_time"), None)]
+        self.reference_cube = iris.cube.Cube(
+            new_data, "air_temperature", units="K",
+            dim_coords_and_dims=dim_coords,
+            aux_coords_and_dims=aux_coords)
+        self.reference_cube.add_aux_coord(AuxCoord(0, "realization"))
+
+        # split into a cubelist for each model
+        self.reference_cubelist = iris.cube.CubeList([self.reference_cube[0],
+                                                      self.reference_cube[1]])
+
+    def test_slices(self):
+        """Test function slices out extra dimensions"""
+        result = self.plugin._slice_input_cubes(self.cube)
+        self.assertIsInstance(result, iris.cube.CubeList)
+        for cube, refcube in zip(result, self.reference_cubelist):
+            self.assertArrayAlmostEqual(cube.data, refcube.data)
+            self.assertEqual(cube.metadata, refcube.metadata)
+
+    def test_cubelist(self):
+        """Test function creates cubelist with same dimensions where needed"""
+        result = self.plugin._slice_input_cubes(self.reference_cube)
+        self.assertIsInstance(result, iris.cube.CubeList)
+        for cube, refcube in zip(result, self.reference_cubelist):
+            self.assertArrayAlmostEqual(cube.data, refcube.data)
+            self.assertEqual(cube.metadata, refcube.metadata)
+
+
 class Test_process(IrisTest):
     """Test the process method"""
 
```

Comment:
Please could you change all instances of "self.plugin_cube" for "self.plugin" in this class.